src/node/transaction.cpp

```diff
@@ -22,7 +22,7 @@ static TransactionError HandleATMPError(const TxValidationState& state, std::str
     err_string_out = state.ToString();
     if (state.IsInvalid()) {
         if (state.GetResult() == TxValidationResult::TX_MISSING_INPUTS) {
-            return TransactionError::MISSING_INPUTS;
```

Comment:
Nice, I think we also rename the one in `TxValidationResult` enum.

g/`TX_MISSING_INPUTS`/`TX_MISSING_INPUT_OR_SPENT`/g
and update comments appropriately.

Reply:
I did consider this, but it felt much less exposed to the user, only via `testmempoolaccept` AFAIK:

```cpp
    result_inner.pushKV("reject-reason", "missing-inputs");
```

Therefore as an "internal error", `TX_MISSING_INPUTS` (for any reason) it felt more accurate to me to leave as is? Very happy to be persuaded away from this viewpoint though :)

Reply:
> Therefore as an "internal error", TX_MISSING_INPUTS (for any reason) it felt more accurate to me to leave as is?

I was reviewing this based on the PR rationale in the OP. You stated that "a transaction's inputs are missing from the UTXO set, which could also be due to all inputs having already been spent (MISSING_INPUTS -> INPUTS_MISSING_OR_SPENT)." That's why I suggested that `TxValidationResult` one  should be changed just for uniformity. But Internally, you're right `TX_MISSING_INPUTS` inferred or spent.

Maybe just update the `testmempoolaccept` reject reason since it's user-facing?

But if you agree, you may have to update some functional tests that use it and maybe a release note since users heavily use the RPC.

This is a non blocking comment.

This PR is an improvement and will close the [issue](https://github.com/bitcoin/bitcoin/issues/19363)